Expensify/App
# Add desktop dock icon indicator in Electron

**Fixes:** https://github.com/Expensify/Expensify/issues/141720

cc @tgolen who probably has some ideas on how to improve this...

In order to accomplish this I added an `unreadActionCount` to reports in addition to `isUnread` this way we can pass a number to anything that wants to know about how many unread there are.

**Other things I'd like to do:**

- ~~Move these `ipcRenderer` events into some kind of `ELECTRON_EVENTS` file~~ Done
- ~~Debounce the `updatePageTitleAndUnreadCount()` method since it gets called many times in quick succession when anything subscribes to `IONKEYS.COLLECTION.REPORT` (maybe there's another way around this - but seems to work ok)~~ Done

**Tests:**

1. Log into the Desktop app
1. Make sure you have one unread chat in your LHN by adding a comment to a report as a shared user
1. Verify your dock icon says there is 1 message
<img width="1200" alt="Screenshot 1376" src="https://user-images.githubusercontent.com/32969087/93937840-37a8c200-fcdd-11ea-8db5-d96008f7ecc2.png">
1. Reboot the app and verify that the icon is updated correctly
1. Click on one of the unread chats 
1. Verify that the dock icon number clears
<img width="1440" alt="Screenshot 1378" src="https://user-images.githubusercontent.com/32969087/93937878-48f1ce80-fcdd-11ea-9552-9a893d50293b.png">
1. Nav away from the chat and add two more messages
1. Verify the dock icon count increments to 2
